projects/core/src/lib/operators/dialogs/raster-kernel/raster-kernel.component.html

```diff
@@ -0,0 +1,116 @@
+                        <mat-icon>settings_suggest</mat-icon>
+                    </button>
+                    <mat-menu #matrixMenu="matMenu">
+                        <button mat-menu-item (click)="presetSobel()">Sobel</button>
+                        <button mat-menu-item (click)="presetGaussianBlur()">Gaussian blur 5x5</button>
+                        <button mat-menu-item (click)="presetMean()">Mean 3x3</button>
+                    </mat-menu>
+                </div>
+                <div class="matrix">
+                    <table>
+                        <tr *ngFor="let row of getMatrix(); index as r">
+                            <td *ngFor="let value of row; index as c">
+                                <mat-form-field appearance="fill">
+                                    <input
+                                        matInput
+                                        type="number"
+                                        [ngModel]="value"
+                                        (ngModelChange)="setMatrixValue(r, c, $event)"
+                                        [ngModelOptions]="{standalone: true}"
+                                    />
+                                </mat-form-field>
+                            </td>
+                        </tr>
+                    </table>
+                </div>
+            </ng-container>
+            <ng-container *ngSwitchCase="'standardDeviation'">
+                <p>Dimensions</p>
+                <div *ngIf="getDimensions() as dimensions" fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="0.5rem">
+                    <mat-form-field appearance="fill" fxFlex>
+                        <mat-label>X</mat-label>
+                        <input
+                            matInput
+                            type="number"
+                            [ngModel]="dimensions[0]"
+                            (ngModelChange)="setDimensionValue(0, $event)"
+                            [ngModelOptions]="{standalone: true}"
+                        />
+                    </mat-form-field>
+                    <span>x</span>
+                    <mat-form-field appearance="fill" fxFlex>
+                        <mat-label>Y</mat-label>
+                        <input
+                            matInput
+                            type="number"
+                            [ngModel]="dimensions[1]"
+                            (ngModelChange)="setDimensionValue(1, $event)"
+                            [ngModelOptions]="{standalone: true}"
+                        />
+                    </mat-form-field>
+                </div>
+            </ng-container>
+        </ng-container>
+
+        <p *ngIf="form.controls.kernel.errors as kernelError" class="error">
+            <span *ngIf="kernelError?.emptyKernel">Dimensions must not be empty</span>
+            <span *ngIf="kernelError?.kernelDimensionsNotOdd">Dimensions must be odd</span>
```

Comment:
if I leave one dimension of the standard deviation matrix empy, this message appears, but it should say empty

Reply:
corrected it in [44d384e](https://github.com/geo-engine/geoengine-ui/pull/476/commits/44d384e6d1ccbc3be47f29601f3b6b351c9d0ef5)